File: javascript/rna-transcription/rna-transcription.js
'use strict'

class DnaTranscriber {
  toRna(strand) {
    let array = Array.from(strand)

    if (array.some(function(x) {
      return !('GCTA'.includes(x))
    })) {
      throw new Error("Invalid input")
    }

    return array.map(function(x) {
      switch (x) {
        case 'G':
          return 'C'
        case 'C':
          return 'G'
        case 'T':
          return 'A'
        case 'A':
          return 'U'
      }
    }).join('')
  }
}

module.exports = DnaTranscriber

File: ruby/rna-transcription/rna_transcription.rb
class Complement
  def self.of_dna(strand)
    array = strand.split('')

    if array.any? { |char| !'GCTA'.include?(char) }
      return ''
    end

    complement = array.map do |char|
      case char
      when 'G'
        'C'
      when 'C'
        'G'
      when 'T'
        'A'
      when 'A'
        'U'
      end
    end

    return complement.join
  end
end
